source/common/config/grpc_mux_impl.cc

```diff
@@ -62,6 +64,13 @@ void GrpcMuxImpl::sendDiscoveryRequest(const std::string& type_url) {
   if (skip_subsequent_node_ && !first_stream_request_) {
     request.clear_node();
   }
+  // TODO(htuch): this works as long as there are no new fields in the v3+
+  // DiscoveryRequest. When they are added, we need to do a full v2 conversion
+  // and also discard unknown fields. Tracked at
+  // https://github.com/envoyproxy/envoy/issues/9619.
+  if (transport_api_version_ == envoy::config::core::v3alpha::ApiVersion::V3ALPHA) {
```

Comment:
Should this potentially be != v2 instead of == v3alpha for when new versions get added? Possibly pedantic, not sure. Up to you.

Reply:
I think we need to come back to this code before too long anyway, per the tracking issue, so I think it's fine for the next short while.